cmd/zfs/zfs_main.c

```diff
@@ -3761,8 +3761,13 @@ collect_dataset(zfs_handle_t *zhp, list_cbdata_t *cb)
 		if (cb->cb_json) {
 			if (pl->pl_prop == ZFS_PROP_NAME)
 				continue;
+			const char *prop_name;
+			if (pl->pl_prop != ZPROP_USERPROP)
+				prop_name = zfs_prop_to_name(pl->pl_prop);
+			else
+				prop_name = pl->pl_user_prop;
```

Comment:
We are assigning `char *` to `const char*`. Though this should not be a problem, I wonder if the compiler can somehow complain.

Reply:
The problem would be opposite -- removing const qualifier. This way is fine.